From 9cb09166096deb6ac07bb77b5a4a9b1840658495 Mon Sep 17 00:00:00 2001
From: ysiyer <yegnesh.s.iyer@intel.com>
Date: Wed, 11 May 2016 14:46:38 +0530
Subject: [PATCH 1/2] EM/mfd_pmic_core: Removing device name from the print in
 irq thread

Issue:
irq thread parses the irq based on the total number of irqs for
different subsystems in pmic (total of thermal, charger, adc, pwrsrc etc).
Multiple irqs are grouped together to their corresponding cell device.
For e.g, thermal might have all the irqs for systherms, pmic thermistor,
battery thermistor all grouped as resouces to the thermal cell device.
Cell devices and irqs have a 1:N mapping. Since, the total irqs always
out number the cell device count, the device name printing will cause
exception.

Solution:
Remove the device name from the pr_info statement and just print the irq
number. User can map the number to the device using cat /proc/interrupts
command in the shell and identify the source of wakeup.

Change-Id: Iba2c64f3bfb3a98912df69cc4d36735aa559b984
Signed-off-by: ysiyer <yegnesh.s.iyer@intel.com>
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-32018
Reviewed-on: https://android.intel.com:443/505893
---
 drivers/mfd/intel_soc_pmic_core.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/mfd/intel_soc_pmic_core.c b/drivers/mfd/intel_soc_pmic_core.c
index 8c9dacc..8e3d057 100644
--- a/drivers/mfd/intel_soc_pmic_core.c
+++ b/drivers/mfd/intel_soc_pmic_core.c
@@ -351,9 +351,8 @@ static irqreturn_t pmic_irq_thread(int irq, void *data)
 			pmic_regmap_write(&pmic->irq_regmap[i].ack,
 				pmic->irq_regmap[i].ack.mask);
 			if (pmic->print_wakeup) {
-				pr_info("pmic: device[%s] irq[%d] might"
+				pr_info("pmic: irq[%d] might"
 					" wake up system!\n",
-					pmic->cell_dev[i].name,
 					pmic->irq_base + i);
 			}
 
-- 
1.9.1

